protobuf/coreapi/api.proto

```diff
@@ -157,9 +157,23 @@ message ListenResultRequest {
 message ResultData {
   string executionID = 1;             // The unique identifier of the execution.
   string taskKey = 2;                 // The key of the executed task.
-  string outputKey = 3;               // The output's key from the returned task.
-  string outputData = 4;              // The output's data from the returned task, encoded in JSON.
   repeated string executionTags = 5;  // The list of tags associated with the execution
+  oneof result {
+    Output output = 6;   // output is filled when there is matching task execution that completed successfully.
+    Error error = 7;     // error is filled when there is a matching task execution that cannot be completed. 
+  }
+
+  message Output {
+    string key = 3;   // The output's key from the returned task.
+    string data = 4;  // The output's data from the returned task, encoded in JSON.
+  }
+  
+  message Error {
+    string message = 1;
+  }
+
+  string outputKey = 3;   // [Deprecated] The output's key from the returned task.
```

Comment:
I would love to remove this, as it will force us to update services, otherwise we will stuck with it for a long time

Reply:
Are you sure it's a breaking change to move it to a oneof message?
```
 message Output {
    string key = 3;   // The output's key from the returned task.
    string data = 4;  // The output's data from the returned task, encoded in JSON.
  }
```

If the number are the same, the client should be able to decode it. But yes, the client will not be compatible with the error (current behavior anyway)

Reply:
It's a breaking change like explained here https://github.com/mesg-foundation/core/pull/596#discussion_r237433781, they're defined in different levels. One of them in the root (outputKey) and other is under another field (output) as _output.key_

Reply:
So are you planning to send both (root and output) to the client for now?

Reply:
Of course, otherwise if we remote the *old ones* from *root* it'd be braking change for other clients.

Reply:
👍